provider/github/diff.go

```diff
@@ -92,23 +108,101 @@ func (d *diffLines) hunks(file string) ([]*hunk, error) {
 	return hunks, nil
 }
 
+var hunkPattern = regexp.MustCompile(`^(@@ -(\d+)(?:,(\d+))? \+(\d+)(?:,(\d+))? @@[^@]*)(?:@@.*|$)`)
+
 func parseHunks(s string) ([]*hunk, error) {
-	i := 0
+	r := strings.NewReader(s)
+	scanner := bufio.NewScanner(r)
+
 	var hs []*hunk
-	for i < len(s) {
-		read, h, err := parseHunk(s[i:])
-		if err != nil {
-			return nil, err
+	var h *hunk
+	var err error
+	var lChunk linesChunk
+	for scanner.Scan() {
+		var lt lineType
+
+		line := scanner.Text()
+		switch true {
+		case strings.HasPrefix(line, "@@"):
+			if lChunk.Lines > 0 {
+				h.Chunks = append(h.Chunks, lChunk)
+			}
+			lChunk = linesChunk{}
+			h, err = parseHunkHeader(line)
+			if err != nil {
+				return nil, err
+			}
+			hs = append(hs, h)
+			continue
+		case strings.HasPrefix(line, "+"):
+			lt = lineAdded
+		case strings.HasPrefix(line, "-"):
+			lt = lineDeleted
+		default:
+			lt = lineContext
 		}
 
-		i += read
-		hs = append(hs, h)
+		if lChunk.Lines != 0 && lChunk.Type != lt {
+			h.Chunks = append(h.Chunks, lChunk)
+			lChunk = linesChunk{}
+		}
+
+		lChunk.Type = lt
+		lChunk.Lines++
+	}
+
+	if err := scanner.Err(); err != nil {
+		return nil, err
+	}
+
+	if lChunk.Lines > 0 {
+		h.Chunks = append(h.Chunks, lChunk)
 	}
 
 	return hs, nil
 }
 
-var hunkPattern = regexp.MustCompile(`^(@@ -(\d+)(?:,(\d+))? \+(\d+)(?:,(\d+))? @@[^@]*)(?:@@.*|$)`)
+func parseHunkHeader(line string) (*hunk, error) {
+	var (
+		err error
+		h   = &hunk{}
+	)
+
+	matches := hunkPattern.FindStringSubmatch(line)
+	if len(matches) == 0 {
+		return nil, fmt.Errorf("bad hunk format")
```

Comment:
Maybe we could do `fmt.Errorf("bad hunk format: %s", line)`? Here and the next ones

Reply:
it will be solved during the resolving of merge conflict. I don't think I should change it in this PR.